### DISCUSSION TITLE
Use `_id` instead of `osid` as field name for auto-generated entity IDs

### DISCUSSION THREAD

**Discussion by gamemaker1** [Sunbird-RC/community#95] [Ideas & Enhancements]
Most databases/services that auto-add ID fields to entities add it as a `_id` field. The name `osid` is confusing and un-intuitive to developers using the registry.

I suggest we use a standard name (e.g., `_id`) for the field that other services (like [MongoDB](https://docs.mongodb.com/manual/core/document/#field-names), [ElasticSearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-id-field.html)) use already, to make the API more intuitive and easy to get started with for developers.

**Comment by pramodkvarma** [comment]
Maybe this can be made configuration with default being the current `osid`. Ensuring backward compatibility to existing large adoptions is key. Let us ensure we don't break upgrades. @dileepbapat any comment on this?

**Comment by dileepbapat** [comment]
Yeah noted, we will make it configurable and use osid for the cases where compatibility is needed.
